ssl/statem/statem_srvr.c

```diff
@@ -3513,6 +3525,77 @@ WORK_STATE tls_post_process_client_key_exchange(SSL_CONNECTION *s,
     return WORK_FINISHED_CONTINUE;
 }
 
+MSG_PROCESS_RETURN tls_process_client_rpk(SSL_CONNECTION *sc, PACKET *pkt)
+{
+    MSG_PROCESS_RETURN ret = MSG_PROCESS_ERROR;
+    SSL_SESSION *new_sess = NULL;
+    EVP_PKEY *peer_rpk = NULL;
+
+    if (!tls_process_rpk(sc, pkt, &peer_rpk)) {
+        /* SSLfatal() already called */
+        goto err;
+    }
```

Comment:
This should be able to set the peer_rpk without matching it, so the verification result could be other than `X509_V_OK`.  The peer_psk should I expect also be part of any serialised session (external session caches, ...), and available on resumption.